nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Updater/UpdaterWorker.cs

```diff
@@ -39,6 +41,9 @@ public async Task RunAsync(string repoRootPath, string workspacePath, string dep
             case ".vbproj":
                 await RunForProjectAsync(repoRootPath, workspacePath, dependencyName, previousDependencyVersion, newDependencyVersion, isTransitive);
                 break;
+            case ".json":
+                await RunForPackagesLockAsync(repoRootPath, workspacePath);
```

Comment:
I don't think we have guarantees about the ordering of dependencies we invoke the native tool with. Meaning this may be invoked prior to updating the project itself.

You could consider walking up from the project file being updated and looking for a package lock to update. You could then invoke restore after the update operation has been completed.

Reply:
I tried that, but in a solution using central package management, it only updates the lock file for one project. Is there a way?

Reply:
This is the code I wrote:
```cs
private async Task RunUpdaterAsync(
        string repoRootPath,
        string projectPath,
        string dependencyName,
        string previousDependencyVersion,
        string newDependencyVersion,
        bool isTransitive)
    {
        ...

        // Update lock file if exists
        if (File.Exists(Path.Combine(Path.GetDirectoryName(projectPath), "packages.lock.json")))
        {
            await LockFileUpdater.UpdateLockFileAsync(repoRootPath, projectPath, _logger);
        }
    }
```

```cs
internal static class LockFileUpdater
{
    public static async Task UpdateLockFileAsync(
        string repoRootPath,
        string projectPath,
        Logger logger)
    {
        var lockPath = Path.Combine(Path.GetDirectoryName(projectPath), "packages.lock.json");
        logger.Log($"  Running for lock file [{Path.GetRelativePath(repoRootPath, lockPath)}]");
        if (!File.Exists(lockPath))
        {
            logger.Log($"    File [{Path.GetRelativePath(repoRootPath, lockPath)}] does not exist.");
            return;
        }

        var (exitCode, stdout, stderr) = await ProcessEx.RunAsync("dotnet", $"restore --force-evaluate {projectPath}");
        if (exitCode != 0)
        {
            logger.Log($"    Lock file update failed.\nSTDOUT:\n{stdout}\nSTDERR:\n{stderr}");
            return;
        }

        logger.Log($"    Saved [{Path.GetRelativePath(repoRootPath, lockPath)}].");
    }
}
```

Reply:
Ah, it was a problem with my test solution.